build-lib/util/licenser.js

```diff
@@ -21,22 +25,20 @@ Licenser.prototype.constructor = Licenser;
 
 Licenser.prototype.build = function () {
 
-  const files = this.inputPaths.reduce((fileAcc, dir) => {
-    const list = fs.readdirSync(dir).map(fileName => {
-      return path.join(dir, fileName);
-    });
-
-    return fileAcc.concat(list);
-  }, []).filter(fileName => {
-    return fileName.match(/^.+\.js$/);
-  });
-
-  files.forEach(fileName => {
-    const inputBuffer = fs.readFileSync(path.join(fileName));
+  this.inputPaths.forEach(dir => {
```

Comment:
i just took a shot at rewriting and might have to scope the file writing while going through the inputs. The `writeFile` bit needs the base directory name from which it was created